lms/views/exceptions.py

```diff
@@ -27,12 +27,23 @@ def notfound(self):
 
     @forbidden_view_config()
     def forbidden(self):
-        if validation_error := getattr(self.exception.result, "validation_error", None):
-            self.request.override_renderer = (
-                "lms:templates/validation_error.html.jinja2"
-            )
-            self.request.response.status_int = 403
-            return {"error": validation_error}
+        if exception := getattr(self.exception.result, "exception", None):
```

Comment:
This is the less satisfactory part of this change

Exception that happen during authentication get here as `reason.exception` and I dispatch them to the right exception view function to treat them in the same way.